components/locale_core/src/preferences/locale.rs

```diff
@@ -20,37 +21,65 @@ pub struct LocalePreferences {
     /// Preference of Variant
     pub(crate) variant: Option<Variant>,
     /// Preference of Regional Subdivision
-    pub(crate) subdivision: Option<SubdivisionId>,
+    pub(crate) subdivision: Option<RegionalSubdivision>,
     /// Preference of Unicode Extension Region
-    pub(crate) ue_region: Option<SubdivisionId>,
+    pub(crate) region_override: Option<RegionOverride>,
 }
 
 impl LocalePreferences {
-    fn to_data_locale_maybe_region_priority(self, region_priority: bool) -> DataLocale {
+    /// Convert to a DataLocale, with region-based fallback priority
+    ///
+    /// Most users should use `icu_provider::marker::make_locale()` instead.
+    pub fn to_data_locale_region_priority(self) -> DataLocale {
+        let (region, subdivision) = if let Some(ro) = self.region_override {
+            // Use the region override if present
+            (
+                Some(ro.region),
+                Some(*ro).filter(|sd| !sd.suffix.is_unknown()),
+            )
+        } else if let Some(sd) = self.subdivision {
+            if let Some(r) = self.region {
+                // Discard the subdivison if it doesn't match the region
+                (
+                    Some(r),
+                    Some(*sd).filter(|sd| sd.region == r && !sd.suffix.is_unknown()),
```

Comment:
where does `sd.region` and `sd.suffix` come from since these are now `RegionalSubdivision` which doesn't have those fields?

Reply:
oh there's a Deref impl. 🤯